paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -281,4 +283,20 @@ public void createTag(String tagName, long fromSnapshotId) {
         TagManager tagManager = new TagManager(fileIO, path);
         tagManager.createTag(snapshot, tagName);
     }
+
+    @Override
+    public void deleteTag(String tagName) {
+        checkArgument(!StringUtils.isBlank(tagName), "Tag name '%s' is blank.", tagName);
+
+        TagManager tagManager = new TagManager(fileIO, path);
+        SnapshotManager snapshotManager = snapshotManager();
+        Snapshot taggedSnapshot = tagManager.taggedSnapshot(tagName);
+
+        if (!snapshotManager.snapshotExists(taggedSnapshot.id())) {
```

Comment:
`if (!snapshotManager.snapshotExists(taggedSnapshot.id())) {`
Put this into `tagDeletion.delete`.